lib/reportDisables.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+const _ = require('lodash');
+
+/** @typedef {import('stylelint').RangeType} RangeType */
+/** @typedef {import('stylelint').DisableReportRange} DisabledRange */
+/** @typedef {import('stylelint').StylelintDisableOptionsReport} StylelintDisableOptionsReport */
+
+/**
+ * Returns a report describing which `results` (if any) contain disabled ranges
+ * for rules that disallow disables via `reportDisables: true`.
+ *
+ * @param {import('stylelint').StylelintResult[]} results
+ * @returns {StylelintDisableOptionsReport}
+ */
+module.exports = function (results) {
+	/** @type {StylelintDisableOptionsReport} */
+	const report = [];
+
+	results.forEach((result) => {
+		// File with `CssSyntaxError` don't have `_postcssResult`s.
+		if (!result._postcssResult) {
+			return;
+		}
+
+		/** @type {{ranges: DisabledRange[], source: string}} */
+		const reported = { source: result.source || '', ranges: [] };
+
+		/** @type {{[ruleName: string]: Array<RangeType>}} */
+		const rangeData = result._postcssResult.stylelint.disabledRanges;
+
+		if (!rangeData) return;
+
+		const config = result._postcssResult.stylelint.config;
+
+		// If no rules actually disallow disables, don't bother looking for ranges
+		// that correspond to disabled rules.
+		if (!Object.values(_.get(config, 'rules', {})).some(reportDisablesForRule)) {
+			return [];
+		}
+
+		Object.keys(rangeData).forEach((rule) => {
+			rangeData[rule].forEach((range) => {
+				if (!reportDisablesForRule(_.get(config, ['rules', rule], []))) return;
+
+				// If an equivalent range has already been reported, don't report this
+				// range as well.
+				const duplicate = reported.ranges.find((reportedRange) => {
```

Comment:
I think this is fine to show duplicate here, because if you are using `reportDisables: true` for rule `N` you want to see report regardless to report of rule `M`

Reply:
Done.